File: 1721-swapping-nodes-in-a-linked-list/1721-swapping-nodes-in-a-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} k
 * @return {ListNode}
 */
const swapNodes = function(head, k) {
	const newHead = new ListNode(0, head);
    
  let slow = newHead, fast = newHead;
  let firstNode = null, secondNode = null;

  while(k > 0) {
      fast = fast.next;
      k -= 1;
  }

  firstNode = fast;

   while(fast) {
      slow = slow.next;
      fast = fast.next;
  }

  secondNode = slow;

  let temp = secondNode.val;
  secondNode.val = firstNode.val;
  firstNode.val = temp;

  return newHead.next;
}

File: 1721-swapping-nodes-in-a-linked-list/1721-swapping-nodes-in-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapNodes(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        
        first = second = None
        fast = slow = head
        
        for i in range(1, k): 
            fast = fast.next
            
        first = fast
        
        while fast.next:
            slow = slow.next
            fast = fast.next
        
        second = slow
        
        first.val, second.val = second.val, first.val
        
        return head 